File: app/src/main/java/lemond/annoying/gamerscompanion/fragment_now/adapter/GameGridViewHolder.java
package lemond.annoying.gamerscompanion.fragment_now.adapter;

import android.support.v7.widget.RecyclerView;

import lemond.annoying.gamerscompanion.databinding.GridItemGameBinding;
import lemond.annoying.gamerscompanion.fragment_now.fragment_main.viewmodel.GameItemViewModel;


public class GameGridViewHolder extends RecyclerView.ViewHolder {

    private GridItemGameBinding binding;

    public GameGridViewHolder(GridItemGameBinding binding) {
        super(binding.getRoot());

        this.binding = binding;
    }

    public GridItemGameBinding getBinding() {
        return binding;
    }

    public void bindItemViewModel(GameItemViewModel viewModel) {
        binding.setViewModel(viewModel);
        binding.executePendingBindings();
    }
}

File: app/src/main/java/lemond/annoying/gamerscompanion/main_activity/fragment_now/adapter/GameGridViewHolder.kt
package lemond.annoying.gamerscompanion.main_activity.fragment_now.adapter

import android.support.v7.widget.RecyclerView

import lemond.annoying.gamerscompanion.databinding.GridItemGameBinding
import lemond.annoying.gamerscompanion.main_activity.fragment_now.fragment_main.viewmodel.GameItemViewModel


class GameGridViewHolder(val binding: GridItemGameBinding) : RecyclerView.ViewHolder(binding.root) {

    fun bindItemViewModel(viewModel: GameItemViewModel) {
        binding.viewModel = viewModel
        binding.executePendingBindings()
    }
}
